multitenant/tenant.go

```diff
@@ -0,0 +1,103 @@
+package multitenant
+
+import (
+	"context"
+	"sync"
+
+	"github.com/centrifuge/go-centrifuge/config"
+	"github.com/centrifuge/go-centrifuge/identity"
+	logging "github.com/ipfs/go-log"
+)
+
+var log = logging.Logger("tenant")
+
+// Config is configuration for a single tenant
+type Config interface {
+	GetEthereumAccount(accountName string) (account *config.AccountConfig, err error)
+	GetEthereumDefaultAccountName() string
+	GetReceiveEventNotificationEndpoint() string
+	GetIdentityID() ([]byte, error)
+	GetSigningKeyPair() (pub, priv string)
+	GetEthAuthKeyPair() (pub, priv string)
+}
+
+// Server interface must be implemented by all tenant specific background servers on Cent Node
+type Server interface {
+
+	// Name is the unique name given to the service within the Cent Node
+	Name() string
+
+	// Start starts the service, expectation is that this would always be called in a separate go routine.
+	// WaitGroup contract must always be honoured by calling `defer wg.Done()`
+	Start(ctx context.Context, wg *sync.WaitGroup, startupErr chan<- error)
+}
+
+// tenant is a tenant in the multi tenant centrifuge node
+type tenant struct {
```

Comment:
Here we will have to add the services exposed in future PRs right?

Reply:
the services are in the bootstrapped context per tenant. so its already in there as you may see.